examples/matrixMultiply.cpp

```diff
@@ -0,0 +1,255 @@
+  RAJA::Range - List of iterables for an index of the loop
+
+  3. RAJA::View - RAJA's wrapper for multidimensional indexing
+
+*/
+int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
+{
+
+  printf("Example 2: Multiplying Two N x N Matrices \n \n");
+  double *A = memoryManager::allocate<double>(NN);
+  double *B = memoryManager::allocate<double>(NN);
+  double *C = memoryManager::allocate<double>(NN);
+
+  for (int i = 0; i < NN; ++i) {
+    A[i] = 1.0;
+    B[i] = 1.0;
+  }
+
+  /*
+    RAJA::View - RAJA's wrapper for multidimensional indexing
+  */
+
+  RAJA::View<double, RAJA::Layout<DIM>> Aview(A, N, N);
+  RAJA::View<double, RAJA::Layout<DIM>> Bview(B, N, N);
+  RAJA::View<double, RAJA::Layout<DIM>> Cview(C, N, N);
+
+  printf("Standard C++ Loop \n");
+  for (int row = 0; row < N; ++row) {
+    for (int col = 0; col < N; ++col) {
+
+      double dot = 0.0;
+      for (int k = 0; k < N; ++k) {
+        dot += Aview(row, k) * Bview(k, col);
+      }
+
+      Cview(row, col) = dot;
+    }
+  }
+  checkSolution<double>(Cview, N);
+
+  /*
+    RAJA bounds may be specified prior to forall statements
+  */
+  RAJA::RangeSegment matBounds(0, N);
+
+  printf("RAJA: Sequential Policy - Single forall \n");
+  RAJA::forall<RAJA::seq_exec>
+    (matBounds, [=](RAJA::Index_type row) {
+  
+    for (int col = 0; col < N; ++col) {
+
+      double dot = 0.0;
+      for (int k = 0; k < N; ++k) {
+        dot += Aview(row, k) * Bview(k, col);
+      }
+      
+      Cview(row, col) = dot;
+    }
+    
+  });
+  checkSolution<double>(Cview, N);
+
+  printf("RAJA: Sequential Policy - Nested forall \n");
+  /*
+    Forall loops may be nested under sequential and omp policies
+  */
+  RAJA::forall<RAJA::seq_exec>
+    (matBounds, [=](RAJA::Index_type row) {  
+      
+      RAJA::forall<RAJA::seq_exec>
+        (matBounds, [=](RAJA::Index_type col) {
+          
+          double dot = 0.0;
+          for (int k = 0; k < N; ++k) {
+            dot += Aview(row, k) * Bview(k, col);
+          }
+          
+          Cview(row, col) = dot;
+        });
+    });
+  checkSolution<double>(Cview, N);
+  
+  
+  printf("RAJA: Sequential Policy RAJA - forallN \n");
+  /*
+    Nested forall loops may be collapsed into a single forallN loop
+  */
+  RAJA::forallN<RAJA::NestedPolicy
+    <RAJA::ExecList<RAJA::seq_exec, RAJA::seq_exec>>>
+    (matBounds, matBounds, [=](RAJA::Index_type row, RAJA::Index_type col) {
+      
+      double dot = 0.0;
+      for (int k = 0; k < N; ++k) {        
+        dot += Aview(row, k) * Bview(k, col);
+      }
+      
+      Cview(row, col) = dot;
+    });
+  checkSolution<double>(Cview, N);
+  
+  
+#if defined(RAJA_ENABLE_OPENMP)
+  printf("RAJA: OpenMP/Sequential Policy - forallN \n");
+  /*
+    Here the outer loop is excuted in parallel while the inner loop
+    is executed sequentially
+  */
+  RAJA::forallN<RAJA::NestedPolicy
+    <RAJA::ExecList<RAJA::omp_parallel_for_exec, RAJA::seq_exec>>>
```

Comment:
I think this is good! If it's in scope, having an example with tiling might be nice. Given that nested::forall is a work in progress I wouldn't invest too much time in forallN examples.